File: python/6_zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        # Check input
        if numRows == 1 or numRows >= len(s):
            return s

        # Initialization
        s_list = [''] * numRows
        pos = 0
        step = 1

        for i, l in enumerate(s):
            s_list[pos] += l
            pos += step

            if pos == 0 or pos == numRows - 1:
                step *= -1

        ans = ''.join(s_list)

        return ans

File: go/6_zigzag-conversion.go
import (
	"strings"
)

func convert(s string, numRows int) string {
	if numRows == 1 {
		return s
	}

	s_list := make([]string, numRows)
	step := 1
	pos := 0

	for i := 0; i < len(s); i++ {
		s_list[pos] += string(s[i])
		pos += step

		if pos == 0 || pos == numRows-1 {
			step *= -1
		}
	}

	ans := strings.Join(s_list, "")
	return ans
}